vendor/src/github.com/docker/libnetwork/drivers/overlay/filter.go

```diff
@@ -12,6 +12,13 @@ const globalChain = "DOCKER-OVERLAY"
 
 var filterOnce sync.Once
 
+var filterChan = make(chan struct{}, 1)
+
+func filterWait() func() {
```

Comment:
This whole thing looks really scary and weird.

Can you please explain this? Why are we doing it?
We are only ever calling the function returned by `filterWait` immediately after getting it. How is this safe for concurrency?